CL-3585 - Show access permissions for surveys even if posting disabled

jamesspeake opened the pull request:
# Changelog

## Fixed
- CL-3585 - Showing access permissions for survey in back office when the survey is closed for submissions

jamesspeake committed:
[CL-3585] Show permissions for surveys even if posting disabled

sebastienhoorens reviewed (changes requested):
We should not add `native_survey?` conditionals.

I think in this case the cause is that somehow posting_enabled was false, while this should never be the case for native surveys as I understand. The code seemed to imply that `posting_enabled` would never be used for native surveys, but that does not seem to be true. Instead, I would propose to modify `participation_context` ensure that posting_enabled is always true in the case of native surveys (by using helper methods of the participation method).

jamesspeake committed:
[CL-3585] Removed native_survey? conditional from the model

jamesspeake commented:
@sebastienhoorens This method actually causes problems because if we make posting_enabled always true for native surveys then the survey will always be open for responses. However, think I've got a solution by adding another helper method.

jamesspeake committed:
[CL-3585] Used guard as made more sense

sebastienhoorens reviewed (changes requested):
My preference would still be to ensure that posting_enabled is always set to true for native surveys. I feel like this solution is more difficult to understand, but can make sense. Perhaps, we could add a comment, saying that all permissions are set to enabled when hide_disabled_permissions? is true?

The hide_disabled_permissions? does not belong in the participation context, and should go to the participation method instead.

jamesspeake committed:
[CL-3585] Refactored to the participation method

jamesspeake commented:
@sebastienhoorens I've moved the method to the participation method and also renamed the function to `return_disabled_actions` which hopefully is clearer and makes more sense within the permissions model where it is used.

sebastienhoorens reviewed (commented):
I think this is good, but we probably also want to add tests for the participation method to make sure that each participation method returns the expected value.

jamesspeake committed:
[CL-3585] Added tests for return_disabled_actions?

jamesspeake commented:
Added tests to all participation methods @sebastienhoorens
